src/store/index.js

```diff
@@ -34,25 +38,76 @@ export default new Vuex.Store({
       state.collections = await state.connection.db.listCollections().toArray()
     },
     async find({ state }, query = {}) {
-      state.totalNumberOfResults = null
+      if (state.collectionRecords[state.collectionName]) {
+        const query_serialization = JSON.stringify(query)
+
+        if (state.collectionRecords[state.collectionName].query === query_serialization) {
+          state.activeRecords = state.collectionRecords[state.collectionName]
+          state.totalNumberOfResults = state.activeRecords.result.length
+          return
+        }
+      }
+
+      state.totalNumberOfResult = null
       const collection = state.connection.db.collection(state.collectionName)
       const cursor = await collection.find(query)
+
       cursor.count((err, result) => {
         if (err) return
-
         state.totalNumberOfResults = result
       })
-      state.records = await cursor.toArray()
+
+      const result = await cursor.toArray()
+
+      state.activeRecords = {
+        query: JSON.stringify(query),
+        result,
+        isEmpty: !result.length
+      }
+
+      state.collectionRecords[state.collectionName] = state.activeRecords
     },
     async setDatabase({ state, dispatch }, databaseName) {
       state.connection = state.connection.useDb(databaseName)
       state.databaseName = databaseName
       state.collectionName = ''
-      state.records = []
+      state.collectionRecords = {}
+      state.openedCollections = []
       await dispatch('getCollections')
     },
     async setCollection({ state }, collectionName) {
       state.collectionName = collectionName
+      state.lastOpenedCollection = collectionName
+
+      if (state.collectionRecords[state.collectionName]) {
+        state.activeRecords = state.collectionRecords[state.collectionName]
+        state.totalNumberOfResults = state.activeRecords.result.length
```

Comment:
I believe we don't need these variables anymore. worst case, let's make them getters in the store to return selected collection's records, etc.